clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -160,6 +157,9 @@ public MemoryRecords build() {
      * Get the max timestamp and its offset. If the log append time is used, then the offset will
      * be either the first offset in the set if no compression is used or the last offset otherwise.
      * @return The max timestamp and its offset
+     *
+     * I don't really understand the logic of this code.
```

Comment:
I did mention in the PR description that this is still work in progress. :) I've been making some notes while reviewing and included this one in a pushed commit by mistake. The code that was replaced by this one special-cased MagicValue_V0 and it was unclear why that was no longer needed. I'll have more coherent questions when I'm finished with this. :)